src/duct/__main__.py

```diff
@@ -13,7 +13,7 @@
 import sys
 import threading
 import time
-from typing import IO, Any, TextIO
+from typing import IO, Any, Generator, TextIO
```

Comment:
[`typing.Generator`](https://docs.python.org/3/library/typing.html#typing.Generator) is deprecated.  Use `collections.abc.Generator` instead.

Reply:
As suggested below, we don't need Generator at all, Iterator is better.